src/lib/commandclass/ColorSwitchCC.ts

```diff
@@ -0,0 +1,446 @@
+	}
+}
+
+interface ColorSwitchCCGetOptions extends CCCommandOptions {
+	colorComponent: ColorComponent;
+}
+
+@CCCommand(ColorSwitchCommand.Get)
+@expectedCCResponse(ColorSwitchCCReport)
+export class ColorSwitchCCGet extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions | ColorSwitchCCGetOptions,
+	) {
+		super(driver, options);
+		if (gotDeserializationOptions(options)) {
+			// TODO: Deserialize payload
+			throw new ZWaveError(
+				`${this.constructor.name}: deserialization not implemented`,
+				ZWaveErrorCodes.Deserialization_NotImplemented,
+			);
+		} else {
+			// Populate properties from options object
+			this._colorComponent = options.colorComponent;
+		}
+	}
+
+	private _colorComponent: ColorComponent;
+	public get colorComponent(): ColorComponent {
+		return this._colorComponent;
+	}
+	public set colorComponent(value: ColorComponent) {
+		if (!ColorComponent[value]) {
+			throw new Error(
+				"colorComponent must be a valid color component index.",
+			);
+		}
+		this._colorComponent = value;
+	}
+
+	public serialize(): Buffer {
+		this.payload = Buffer.allocUnsafe(1);
+		this.payload[0] = this._colorComponent;
+		return super.serialize();
+	}
+}
+
+export interface ColorSwitchCCSetOptions extends CCCommandOptions, ColorTable {
+	duration?: number;
+}
+
+// TODO: What does this return?  Does it not have an expected response?
+@CCCommand(ColorSwitchCommand.Get)
+export class ColorSwitchCCSet extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions | ColorSwitchCCSetOptions,
+	) {
+		super(driver, options);
+		if (gotDeserializationOptions(options)) {
+			// TODO: Deserialize payload
+			throw new ZWaveError(
+				`${this.constructor.name}: deserialization not implemented`,
+				ZWaveErrorCodes.Deserialization_NotImplemented,
+			);
+		} else {
+			// Populate properties from options object
+			this._colorTable = {
+				warmWhite: options.warmWhite,
+				coldWhite: options.coldWhite,
+				red: options.red,
+				green: options.green,
+				blue: options.blue,
+				cyan: options.cyan,
+				purple: options.purple,
+				index: options.index,
+			};
+			this.duration = options.duration || 0;
+		}
+	}
+
+	private _colorTable: ColorTable;
+	public get colorTable(): ColorTable {
+		return this._colorTable;
+	}
+	public set colorTable(value: ColorTable) {
+		this._colorTable = {
+			warmWhite: value.warmWhite,
+			coldWhite: value.coldWhite,
+			red: value.red,
+			green: value.green,
+			blue: value.blue,
+			cyan: value.cyan,
+			purple: value.purple,
+			index: value.index,
+		};
+	}
+
+	private _duration: number = 0;
+	public get duration(): number {
+		return this._duration;
+	}
+	public set duration(value: number) {
+		if (value < 0 || value > 255) {
+			throw new Error("duration out of range.");
+		}
+		this._duration = value;
+	}
+
+	public serialize(): Buffer {
+		let colorComponentKeys = Object.keys(
```

Comment:
Ok, now I see why you're doing the stuff at the start. There's probably a way to do a bit simpler, but that seems fine for now.